### PR TITLE
added tests around single channel in session and sessionless requests

### PR DESCRIPTION
**Pull request by PommeVerte**
I added the tests I had around this change. These pass:

```
mvn clean install -DincludeNeo4j
mvn clean -DskipIntegrationTests=false -DincludeNeo4j --projects gremlin-server verify
```

The driver end tests pass perfectly. It's looking good.

### CODE REVIEW

File: gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinDriverIntegrateTest.java [hunk 1]
```diff
@@ -102,6 +102,10 @@ public Settings overrideSettings(final Settings settings) {
                 deleteDirectory(new File("/tmp/neo4j"));
                 settings.graphs.put("graph", "conf/neo4j-empty.properties");
                 break;
+            case "shouldExecuteInSessionAndSessionlessWithoutOpeningTransaction":
```
Comment (spmallette) [inline gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinDriverIntegrateTest.java]: @PommeVerte note that you can fold this up into the `case` above as the neo4j initialization code is the same.  basically:

```java
case "shouldExecuteScriptInSessionOnTransactionalGraph":
case "shouldExecuteSessionlessScriptOnTransactionalGraph":
case "shouldExecuteScriptInSessionOnTransactionalWithManualTransactionsGraph":
case "shouldExecuteInSessionAndSessionlessWithoutOpeningTransaction":
    deleteDirectory(new File("/tmp/neo4j"));
    settings.graphs.put("graph", "conf/neo4j-empty.properties");
    break;
````

**Comment by PommeVerte** [comment]
dur. It's a slow monday. I've made the change

**Comment by okram** [comment]
The code looks clean and simple to me.

VOTE: +1

**Comment by spmallette** [comment]
I've run these tests locally when I added the code that I did in this branch.  `SimpleClient` is a wreck in this kind of test case, but not much we can do about that for now. Hopefully we can get some improvements there at some point to simplify that test a bit.

VOTE: +1

**Comment by spmallette** [comment]
It just occurred to me that we don't need a VOTE here.  This is a PR to a branch that is not a "release" branch - I can just merge this.  Then I will make the PR to master where we will need a VOTE.

**Comment by PommeVerte** [comment]
I requested the merge against TINKERPOP3-910 (and not master) does this have to go through a double vote? one here and then in 910?

**Pull request merged**